### PR TITLE
[SPARK-20830][PYSPARK][SQL] Add posexplode and posexplode_outer

### PR DESCRIPTION
**Pull request by zero323**
## What changes were proposed in this pull request?

Add Python wrappers for `o.a.s.sql.functions.explode_outer` and `o.a.s.sql.functions.posexplode_outer`.

## How was this patch tested?

Unit tests, doctests.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @ueshin

File: python/pyspark/sql/tests.py [hunk 1]
```diff
@@ -272,6 +276,11 @@ def test_explode(self):
         self.assertEqual(result[0][0], "a")
         self.assertEqual(result[0][1], "b")
 
+        self.assertEqual(data.select(posexplode_outer("intlist")).count(), 5)
```
Comment (ueshin) [inline python/pyspark/sql/tests.py]: Can you extract the tests like ones for `explode`?

**Review by ueshin — COMMENTED** [review]
LGTM, except for a comment.

File: python/pyspark/sql/tests.py (hunk 1 above)
Comment (zero323) [inline python/pyspark/sql/tests.py]: @ueshin Of course, is this enough?

Comment (ueshin) [inline python/pyspark/sql/tests.py]: Thanks, looks good.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Comment by zero323** [comment]
Thanks @ueshin!

**Pull request closed without merging**